### PR TITLE
Operation helpers

### PR DESCRIPTION
**Pull request by zsoltk**
## Description

Some helpers to make creating new NavModels easier.

## Check list

- [x] I have updated `CHANGELOG.md` if required.
- [x] I have updated documentation if required.

### CODE REVIEW

File: navmodel-samples/src/main/kotlin/com/bumble/appyx/navmodel/tiles/operation/DeselectAll.kt [hunk 1]
```diff
@@ -12,21 +14,10 @@ class DeselectAll<T : Any> : TilesOperation<T> {
...
-                )
-            } else {
-                it
-            }
+    ): NavElements<T, Tiles.State> =
+        elements.transitionTo(STANDARD) {
+            it.targetState == SELECTED
         }
-
-    override fun equals(other: Any?): Boolean = this.javaClass == other?.javaClass
```
Comment (LachlanMcKee) [inline navmodel-samples/src/main/kotlin/com/bumble/appyx/navmodel/tiles/operation/DeselectAll.kt:27]: This isn't a data class. Are you sure we should delete this?

Comment (zsoltk) [reply navmodel-samples/src/main/kotlin/com/bumble/appyx/navmodel/tiles/operation/DeselectAll.kt:27]: No, it was most probably accidental removal.

File: core/src/main/kotlin/com/bumble/appyx/core/navigation/Operation.kt [hunk 2]
```diff
@@ -8,6 +8,57 @@ interface Operation<NavTarget, State> :
 
     fun isApplicable(elements: NavElements<NavTarget, State>): Boolean
 
+    /**
+     * Transitions all elements to a new state
+     */
+    fun NavElements<NavTarget, State>.transitionTo(
+        newTargetState: State
+    ) = transitionToIndexed(newTargetState) { _, _ -> true }
```
Comment (LachlanMcKee) [inline core/src/main/kotlin/com/bumble/appyx/core/navigation/Operation.kt:16]: This creates an unnecessary object, and also doesn't need to access the index. I would recommend just mapping and calling

```
element.transitionTo(
                newTargetState = newTargetState,
                operation = this@Operation
            )
```

It might be slightly more repetitive code wise, but it's better for the end user

File: core/src/main/kotlin/com/bumble/appyx/core/navigation/Operation.kt [hunk 3]
```diff
@@ -8,6 +8,57 @@ interface Operation<NavTarget, State> :
...
+            newTargetState = newTargetState.invoke(element),
+            operation = this@Operation
+        )
+    }
+
+    /**
+     * Transitions elements to a new state if they pass a condition based on the element
+     */
+    fun NavElements<NavTarget, State>.transitionTo(
+        newTargetState: State, condition: (element: NavElement<NavTarget, State>) -> Boolean
```
Comment (LachlanMcKee) [inline core/src/main/kotlin/com/bumble/appyx/core/navigation/Operation.kt:34]: second argument on a new line?

File: core/src/main/kotlin/com/bumble/appyx/core/navigation/Operation.kt [hunk 4]
```diff
@@ -8,6 +8,57 @@ interface Operation<NavTarget, State> :
...
+        } else {
+            element
+        }
+    }
+
+    /**
+     * Transitions elements to a new state if they pass a condition based on the element and its position
+     */
+    fun NavElements<NavTarget, State>.transitionToIndexed(
+        newTargetState: State, condition: (index: Int, element: NavElement<NavTarget, State>) -> Boolean
```
Comment (LachlanMcKee) [inline core/src/main/kotlin/com/bumble/appyx/core/navigation/Operation.kt:50]: second argument on a new line?

Comment (LachlanMcKee) [inline core/src/main/kotlin/com/bumble/appyx/core/navigation/Operation.kt:50]: Does the condition function need named arguments? In the PR it looks like it could cause confusion with potential variable name shadowing

File: navmodel-samples/src/main/kotlin/com/bumble/appyx/navmodel/tiles/transitionhandler/TilesTransitionHandler.kt [hunk 5]
```diff
@@ -62,7 +63,7 @@ class TilesTransitionHandler<T>(
 
 @Composable
 fun <T> rememberTilesTransitionHandler(
-    transitionSpec: TransitionSpec<Tiles.State, Float> = { spring() }
+    transitionSpec: TransitionSpec<Tiles.State, Float> = { spring(stiffness = Spring.StiffnessVeryLow) }
```
Comment (LachlanMcKee) [inline navmodel-samples/src/main/kotlin/com/bumble/appyx/navmodel/tiles/transitionhandler/TilesTransitionHandler.kt:66]: is this related to the PR?

Comment (zsoltk) [reply navmodel-samples/src/main/kotlin/com/bumble/appyx/navmodel/tiles/transitionhandler/TilesTransitionHandler.kt:66]: Not strictly. I changed it while testing, as the default spring makes it very difficult to see what's happening.
